lib/internal/test_runner/mock/mock.js

```diff
@@ -651,7 +651,7 @@ function setupSharedModuleState() {
     const mockExports = new SafeMap();
     const { port1, port2 } = new MessageChannel();
 
-    register('node:test/mock_loader', {
+    register('node:internal/test_runner/mock_loader', {
```

Comment:
You're missing a --expose-internals in test/parallel/test-runner-module-mocking.js

Reply:
That would not be a correct fix, we do not want users to pass `--expose-internals` to use that feature. I thought the fix had landed with the type-stripping PR, but I guess I misremembered.